### PR TITLE
Drop deprecated APIs

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -8,6 +8,20 @@ awareness about deprecated code.
 
 # Upgrade to 3.0
 
+## BC Break: removed `AnnotationDriver`
+
+It has been deprecated in favor of `ColocatedMappingDriver`.
```
Comment (malarzm) [inline UPGRADE.md]: s/deprecated/removed everywhere I think :)

File: tests/Persistence/Mapping/ClassMetadataFactoryTest.php [hunk 2]
```diff
@@ -49,15 +50,15 @@ public function testSetGetCache(): void
...
 
-        self::assertTrue($this->cmf->hasMetadataFor('stdClass'));
+        self::assertTrue($this->cmf->hasMetadataFor(stdClass::class));
     }
 
     public function testGetMetadataForAbsentClass(): void
     {
         $this->expectException(MappingException::class);
-        $this->cmf->getMetadataFor(__NAMESPACE__ . '\AbsentClass');
+        $this->cmf->getMetadataFor(Foo::class);
```
Comment (malarzm) [inline tests/Persistence/Mapping/ClassMetadataFactoryTest.php:61]: Staying with the former wouldn't produce a new PHPStan's baseline entry

**Review by malarzm — DISMISSED** [review]

File: tests/Persistence/Mapping/ClassMetadataFactoryTest.php (hunk 2 above)
Comment (greg0ire) [reply tests/Persistence/Mapping/ClassMetadataFactoryTest.php:61]: In isolation, yes, but there are other occurrences of `Foo::class` in that file.

File: UPGRADE.md (hunk 1 above)
Comment (greg0ire) [reply UPGRADE.md]: What I mean is: we removed `AnnotationDriver` because it has been deprecated in favor of `ColocatedMappingDriver`.

Comment (malarzm) [reply UPGRADE.md]: Ah, I've read that the other way. Anyway it's inconsistent with next paragraphs this way :)

Comment (greg0ire) [reply UPGRADE.md]: I used another phrasing 👍

**Review by malarzm — APPROVED** [review]

**Pull request merged**